genai/example_test.go

```diff
@@ -50,6 +50,10 @@ func uploadFile(ctx context.Context, client *genai.Client, path string) (*genai.
 	}
 	defer osf.Close()
 
+	// The service detects the MIME type of uploaded data automatically, but
+	// sometimes we have to specify it manually. One example is when text/* MIME
+	// types are considered (as it may be difficult to automatically detect the
+	// exact sub-type of a text file).
 	opts := &genai.UploadFileOptions{}
 	if filepath.Ext(path) == ".txt" {
 		opts.MIMEType = "text/plain"
```

Comment:
Thanks, Eli!

Thinking about this a bit more, I'm not sure canonical sample code should have a hard-coded `.txt` -> `text/plain` mapping:

If this is to workaround a current issue where 1) the service applies the `text/plain; charset=utf-8` MIME type to text files when one isn't specified during upload, and 2) the uploaded file cannot be used to generate content because of the `; charset=utf-8` suffix, then we should instead: a) (ideally) fix the backend service, or (less great alternatives) b) change the SDK API to require a MIME type, or c) change the SDK so that this 'workaround' is part of the implementation

—
Thinking about MIME types in the context of canonical sample code, I wonder if the right thing to do here would be:
1. Have the `uploadFile` helper function take an optional MIME type as argument
2. Have most of the uses of this function pass `""` (automatic determination by the server)
3. In `text/*` use cases, pass in an explicit MIME type.
A comment about MIME types is probably still worthwhile, especially about ones that are unlikely to be detected correctly in all cases, which is probably true for `text/*` MIME types that aren't intended to be `text/plain`. Suggested comment:

// To

Reply:
Your suggested comment got cut off

Regarding the `text/plain` workaround - yes. I'm hoping they fix the service soon. Unfortunately, fixing it on the SDK side without breaking API would be challenging and I wouldn't want to take drastic measures just to work around a temporary bug - this is where documentation can be helpful. Agree that moving the MIME setup out of `uploadFile` and documenting it just in the relevant example makes more sense

Reply:
I'll go ahead and back off this comment from this PR for now, focusing it just on one thing (adding audio samples).

I'll send a separate PR with a more meaningful refactor of `uploadFile` that will let us add comments/workarounds in the right place

Reply:
Thanks again.

Apologies for the confusing trailing `// To` at the end of that comment.

Reply:
Fred, please see https://github.com/google/generative-ai-go/pull/185